### ISSUE TITLE
InjectionPoint autowiring throws exception for @Resource beans autowired by name

### ISSUE THREAD

**Issue by undefined-sin** [spring-projects/spring-framework#22359]
I'm not sure if this is the desired behavior, but accessing the `InjectionPoint` of a **named** resource bean will throw the exception below, in version(`5.1.4`).
```
 No current InjectionPoint available for method 'myService' parameter 0.
```
**Example code:** 
```java
@Configuration
public class BeanConfig {

        @Resource
        // @Resource(name = "myService") -> also causes the exception.
        private MyService myService;


        @Bean
        @Scope(BeanDefinition.SCOPE_PROTOTYPE)
        MyService myService(InjectionPoint p) {
            return new MyService();
        }
    }
```
Renaming either the _private field_ to a value different than the bean name will allow the container to inject the `InjectionPoint` bean as expected, because the autowiring strategy fallsback to by type. 
```java
@Configuration
public class BeanConfig {

   @Resource
   private MyService myService;

   @Bean
   @Scope(BeanDefinition.SCOPE_PROTOTYPE)
   MyService myService2(InjectionPoint p) {
        return new MyService();
   }
}
```
**Affects:** 5.1.4

**Comment by jhoeller** [comment]
Indeed, `InjectionPoint` support is currently just available for actual autowiring, not for straight by-name lookups as performed for `@Resource(name = ...)`. I'll revisit this for 5.1.5, exposing the injection point for that name-based lookup variant as well.

**Comment by jhoeller** [comment]
This is available in `5.1.5.BUILD-SNAPSHOT` from https://repo.spring.io/snapshot now. Feel free to give it an early try before the official 5.1.5 release on Wednesday...

**Comment by undefined-sin** [comment]
@jhoeller I just tried and it works perfectly, thanks!
